src/TypeLevel/RowList.purs

```diff
@@ -0,0 +1,61 @@
+module TypeLevel.RowList
+  ( class AllUniqueLabels
+  , tests
+  ) where
+
+import Prim.TypeError (class Fail, Text)
+import Type.RowList (Cons, Nil, RowList)
+
+-- | Uniqueness constraint on the labels of an unordered RowList
+class AllUniqueLabels :: forall (k :: Type). RowList k -> Constraint
+class AllUniqueLabels list
+
+instance AllUniqueLabels Nil
+instance AllUniqueLabels (Cons k a Nil)
+instance
+  Fail (Text "Labels are not unique!") =>
+  AllUniqueLabels (Cons k a (Cons k a' xs))
+else instance
+  ( AllUniqueLabels (Cons k a xs)
+  , AllUniqueLabels (Cons k' a' xs)
+  ) =>
+  AllUniqueLabels (Cons k a (Cons k' a' xs))
+
+-- | Poor man's type level tests
+tests ∷ Array String
+tests =
+  [ testNil
+  , testSingleton
+  , testUniques
+  -- , testDupsUnordered
```

Comment:
Why are these commented out? I see the definitions aren't

Reply:
@bladyjoker implemented these tests so he should probably say whether he thinks the commented test is still needed, but I'll move everything to a more appropriate module in the next set of commits

Reply:
Oh now I remember: Drazen missed an instance when reviewing and thought the compile-time checks worked a bit differently than they actually do. Nothing with duplicates should/does compile. I'm going to remove those.

Reply:
The problem with type level programming, is that you can't test your type level programs :)